functions/src/services/GoogleSheet.ts

```diff
@@ -0,0 +1,358 @@
+    // console.log("Range query: ", sheet + this._computeRange(limit));
+    const targetSheet: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: sheetName + this._computeRange(limit),
+    });
+
+    const { statusText, status, data, headers } = targetSheet;
+    if (statusText !== 'OK' || status !== 200) {
+      console.error('Error: Not able to get google sheet');
+      return null;
+    }
+    const { majorDimension, values }: IValueRange = data;
+    if (majorDimension !== IMajorDimensions.Rows || !values || !values.length) {
+      console.error('Error: Values are wrong format');
+      return null;
+    }
+
+    const headerKeys: any[] = values.shift();
+    return {
+      headers: headerKeys,
+      rows: this._convertRows(headerKeys, values),
+    };
+  }
+
+  public async getColumn(
+    spreadsheetId: ISheetId,
+    sheetName: IProjectSpreadSheetNames,
+    columnName: string
+  ) {
+    await this.connect();
+
+    // Retrieve first row of headers
+    const firstRow: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: sheetName + '!A1:Z1',
+    });
+
+    const { statusText, status, data } = firstRow;
+    if (statusText !== 'OK' || status !== 200) {
+      console.error('Error: Not able to get google sheet');
+      return null;
+    }
+
+    const { values }: IValueRange = data;
+    let targetedColumn;
+    const headers: any[] = values[0];
+    headers.forEach((elem, index) => {
+      if (elem === columnName) {
+        targetedColumn = letters[index];
+      }
+    });
+
+    // console.log("Column letter of File Name: ", targetedColumn);
+
+    const entireColumn: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: sheetName + `!${targetedColumn}1:${targetedColumn}`,
+    });
+    return [].concat.apply([], entireColumn.data.values);
+  }
+
+  /**
+   * Updates a specific Allotment Sheet row
+   *
+   * @param sheetId Update row with known row number
+   * @param sheetName
+   * @param appendValues
+   */
+  public async updateAllotmentRow(
+    sheetId: string,
+    sheetName: IProjectSpreadSheetNames,
+    rowNumber: number,
+    updateValues: any
+  ): Promise<any> {
+    const targetedRange = sheetName + `!A${rowNumber}:Z${rowNumber}`;
+    // Get our targeted row
+    const targetRow: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId: sheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: targetedRange,
+    });
+
+    // Update this row with new merged data from database
+    // Object.keys(updateValues).forEach((elem, index) => {
+    //   targetRow.data.values[0][index] = targetRow.data.values[0][index];
+    // });
+    const resource = {
+      values: [
+        [
+          targetRow.data.values[0][0], // "Days passed"
```

Comment:
Is it possible to update only certain cells without updating the whole row with current values like this?

Reply:
It doesn't seem like there's a way to specifically target cells by column name, so would likely need to calculate specific ranges for each individual cell if we want to do this without repopulating existing values.

Reply:
OK, no problem, no need to spend more time on this.